2.1.0: getrusage misdeclared locally on non-linux architectures

This is a forward of [Debian bug #830253](https://bugs.debian.org/830253):

Builds of casacore on [kFreeBSD](https://buildd.debian.org/status/fetch.php?pkg=casacore&arch=kfreebsd-amd64&ver=2.1.0-1&stamp=1467844627) and the [Hurd](https://buildd.debian.org/status/fetch.php?pkg=casacore&arch=hurd-i386&ver=2.1.0-1&stamp=1467848158) failed with errors along the lines of

```
  /usr/include/i386-kfreebsd-gnu/sys/resource.h:87:69: error: declaration of 'int getrusage(__rusage_who_t, rusage*) throw ()' has a different exception specifier
   extern int getrusage (__rusage_who_t __who, struct rusage *__usage) __THROW;
                                                                       ^
  In file included from /«PKGBUILDDIR»/casacore/casa/OS/MemoryTrace.h:32:0,
                   from /«PKGBUILDDIR»/casa/Containers/Block.cc:29:
  /«PKGBUILDDIR»/casacore/casa/OS/Timer.h:58:16: error: from previous declaration 'int getrusage(int, rusage*)'
   extern "C" int getrusage(int, struct rusage*);
                  ^
  casa/CMakeFiles/casa_casa.dir/build.make:701: recipe for target 'casa/CMakeFiles/casa_casa.dir/Containers/Block.cc.o' failed
```

To address this error, I would suggest editing lines 44 and 192 of `casa/OS/Timer.h` and lines 47, 78, 118, 148, 183, 223, 254, and 290 of `casa/OS/Timer.cc` to check `__GLIBC__` either instead of or in addition to `AIPS_LINUX`.  (An even better fix would be to have the build system check whether a prototype is available.)  There may well be references to `AIPS_LINUX` in other files that could also stand to be generalized.

--- a/casa/OS/Timer.cc
+++ b/casa/OS/Timer.cc
@@ -44,7 +44,7 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
 #if defined (DOS) || defined (MSDOS)
     usage0 = clock();
     ftime(&real0);
-#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD)
+#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD) || defined(__GLIBC__)
 # ifdef AIPS_CRAY_PGI
     struct timezone tz;
     getrusage(0, &usage0);
@@ -75,7 +75,7 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
     ms = ms * 0.001 + s;
     return (ms);
  
-#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD)
+#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD) || defined(__GLIBC__)
 # ifdef AIPS_CRAY_PGI
     struct timeval now;
     struct timezone tz;
@@ -115,7 +115,7 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
     // error: Processor time not available
     return (0.0);
  
-#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD)
+#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD) || defined(__GLIBC__)
 # ifdef AIPS_CRAY_PGI
     double dsec, dusec;
     rusage usage1;       // current rusage structure
@@ -145,7 +145,7 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
 #if defined (DOS) || defined (MSDOS)
     return(0L);
 
-#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD)
+#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD) || defined(__GLIBC__)
 # ifdef AIPS_CRAY_PGI
     double dsec, dusec;
     rusage usage1;       // current rusage structure
@@ -180,7 +180,7 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
     // error: Processor time not available
     return (0.0);
  
-#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD)
+#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD) || defined(__GLIBC__)
 # ifdef AIPS_CRAY_PGI
     double dsec, dusec;
     rusage usage1;       // current rusage structure
@@ -220,7 +220,7 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
     // error: Processor time not available
     return (0.0);
  
-#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD)
+#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD) || defined(__GLIBC__)
 # ifdef AIPS_CRAY_PGI
     double dsec, dusec;
     rusage usage1;       // current rusage structure
@@ -251,7 +251,7 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
 #if defined (DOS) || defined (MSDOS)
     return(0.0);
 
-#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD)
+#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD) || defined(__GLIBC__)
 # ifdef AIPS_CRAY_PGI
     double dsec, dusec;
     rusage usage1;       // current rusage structure
@@ -287,7 +287,7 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
     // error: Processor time not available
     return (0.0);
  
-#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD)
+#elif defined (AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD) || defined(__GLIBC__)
 # ifdef AIPS_CRAY_PGI
     double dsec, dusec;
     rusage usage1;       // current rusage structure

--- a/casa/OS/Timer.h
+++ b/casa/OS/Timer.h
@@ -41,7 +41,7 @@ extern "C" {
 #include <time.h>
 }
 
-#elif defined(AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD)
+#elif defined(AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD) || defined(__GLIBC__)
   #if defined(AIPS_CRAY_PGI)
     #include <sys/time.h>
     #include <sys/resource.h>
@@ -189,7 +189,7 @@ private:
 #if defined(DOS) || defined(MSDOS)
     clock_t usage0;
     timeb   real0;          //# elapsed real time at last mark
-#elif defined(AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD)
+#elif defined(AIPS_SOLARIS) || defined(AIPS_IRIX) || defined(AIPS_OSF) || defined(__hpux__) || defined(AIPS_LINUX) || defined(AIPS_DARWIN) || defined(AIPS_BSD) || defined(__GLIBC__)
   #if defined(AIPS_CRAY_PGI)
     //struct timeval usage0;
     rusage usage0;          //# rusage structure at last mark

